saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/RelyingPartyRegistration.java

```diff
@@ -639,15 +657,32 @@ public Builder singleSignOnServiceBinding(Saml2MessageBinding singleSignOnServic
 				return this;
 			}
 
+			/**
+			 * Apply this {@link Consumer} to the list of signature algorithm URIs
+			 * @param signingMethodAlgorithmsConsumer a {@link Consumer} of the list of
+			 * signature algorithm URIs
+			 * @return this {@code Builder}
+			 * @since 5.5
+			 */
+			public Builder signingMethodAlgorithms(Consumer<List<String>> signingMethodAlgorithmsConsumer) {
+				signingMethodAlgorithmsConsumer.accept(this.signingMethodAlgorithms);
+				return this;
+			}
+
 			/**
 			 * Creates an immutable ProviderDetails object representing the configuration
 			 * for an Identity Provider, IDP
 			 * @return immutable ProviderDetails object
 			 */
 			public AssertingPartyDetails build() {
+				if (this.signingMethodAlgorithms.isEmpty()) {
+					this.signingMethodAlgorithms.add(SignatureConstants.ALGO_ID_SIGNATURE_RSA_SHA256);
+				}
+				List<String> signingMethodAlgorithmsCopy = new ArrayList(this.signingMethodAlgorithms);
+				this.signingMethodAlgorithms.clear();
```

Comment:
What is your reasoning for clearing the list out? `build` does not clear out the other lists or reset any of the other fields to their original state.

Reply:
Good point, It was to solve the initialization issue where default algo was always there because of the fact that a default AssertingPartyDetails is created with RelyingPartyRegistration instance but after looking at this more carefully, it seems the wrong way to fix this, I will improve it